tests/utils/hash_test_suite.cpp

```diff
@@ -0,0 +1,215 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE
+//
+// @license: http://www.opensource.org/licenses/mit-license.php
+// @author: see AUTHORS file
+
+#include <helpers/test_suite.hpp>
+#include <helpers/timing.hpp>
+
+#include <xlnt/styles/color.hpp>
+#include <xlnt/styles/font.hpp>
+#include <xlnt/utils/hash.hpp>
+
+#include <chrono>
+#include <iostream>
+#include <random>
+#include <unordered_set>
+#include <vector>
+
+class hash_test_suite : public test_suite
+{
+public:
+    hash_test_suite()
+    {
+        register_test(test_color_hash_functionality);
+        register_test(test_font_hash_functionality);
+        register_test(test_color_hash_performance);
+        register_test(test_font_hash_performance);
+        register_test(test_hash_consistency);
+    }
+
+    void test_color_hash_functionality()
+    {
+        // Test that the same color produces the same hash
+        xlnt::color color1 = xlnt::color::red();
+        xlnt::color color2 = xlnt::color::red();
+        std::hash<xlnt::color> hasher;
+
+        xlnt_assert_equals(hasher(color1), hasher(color2));
+
+        // Test that different colors produce different hashes
+        xlnt::color color3 = xlnt::color::blue();
+        xlnt_assert(hasher(color1) != hasher(color3));
+
+        // Test indexed colors
+        xlnt::color indexed1(xlnt::indexed_color(1));
+        xlnt::color indexed2(xlnt::indexed_color(2));
+        xlnt_assert(hasher(indexed1) != hasher(indexed2));
+
+        // Test theme colors
+        xlnt::color theme1(xlnt::theme_color(1));
+        xlnt::color theme2(xlnt::theme_color(1));
+        xlnt_assert_equals(hasher(theme1), hasher(theme2));
+
+        // Test tinted colors
+        xlnt::color tinted = xlnt::color::red();
+        tinted.tint(0.5);
+        xlnt_assert(hasher(color1) != hasher(tinted));
+    }
+
+    void test_font_hash_functionality()
+    {
+        std::hash<xlnt::font> hasher;
+
+        // Test default font
+        xlnt::font font1;
+        xlnt::font font2;
+        xlnt_assert_equals(hasher(font1), hasher(font2));
+
+        // Test fonts with different names
+        xlnt::font font3;
+        font3.name("Arial");
+        xlnt_assert(hasher(font1) != hasher(font3));
+
+        // Test fonts with different sizes
+        xlnt::font font4;
+        font4.size(14.0);
+        xlnt_assert(hasher(font1) != hasher(font4));
+
+        // Test bold fonts
+        xlnt::font font5;
+        font5.bold(true);
+        xlnt_assert(hasher(font1) != hasher(font5));
+
+        // Test fonts with colors
+        xlnt::font font6;
+        font6.color(xlnt::color::red());
+        xlnt_assert(hasher(font1) != hasher(font6));
+    }
+
+    void test_color_hash_performance()
+    {
+        const int test_size = 10000;
```

Comment:
As doomlaur also noticed, this is rather a benchmark than a unit test. I think this benchmark is also not very relevant, as it tests the performance of the unordered_map implementation rather than our hashing function.

What we could test instead is if all those hashes are unique, but I would use some fixed values (e.g. all integers between 0 and 255 with a step of 10) for it instead of a random generator to avoid this unit test from failing randomly.